### PR TITLE
Support reloading storage data when gcs server restarts

### PR DESCRIPTION
**Pull request by ffbin**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_server.cc [hunk 1]
```diff
@@ -95,12 +95,23 @@ void GcsServer::Start() {
...
+  auto load_completed_count = std::make_shared<int>(0);
+  int load_count = 3;
+  auto on_done = [this, load_count, load_completed_count](const Status &status) {
+    ++(*load_completed_count);
 
-  // Store gcs rpc server address in redis.
-  StoreGcsServerAddressInRedis();
-  is_started_ = true;
+    if (*load_completed_count == load_count) {
+      // Run rpc server.
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/gcs_server.cc:104]:
```suggestion
      // Start RPC server when all tables have finished loading initial data.
```

File: src/ray/gcs/gcs_server/gcs_object_manager.h [hunk 2]
```diff
@@ -54,6 +54,11 @@ class GcsObjectManager : public rpc::ObjectInfoHandler {
                                   rpc::RemoveObjectLocationReply *reply,
                                   rpc::SendReplyCallback send_reply_callback) override;
 
+  /// Load data from gcs storage to memory cache asynchronously.
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/gcs_object_manager.h:57]:
```suggestion
  /// Load initial data from gcs storage to memory cache asynchronously.
  ///
  /// This should be called when GCS server restarts after a failure.
```

Comment (raulchen) [reply src/ray/gcs/gcs_server/gcs_object_manager.h:57]: same for other managers.

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 3]
```diff
@@ -721,5 +721,36 @@ void GcsActorManager::SchedulePendingActors() {
...
+        registered_actors_.emplace(item.first, actor);
+
+        if (actor->IsDetached()) {
+          named_actors_.emplace(actor->GetName(), actor->GetActorID());
+        }
+
+        created_actors_[actor->GetNodeID()].emplace(actor->GetWorkerID(),
+                                                    actor->GetActorID());
+
+        auto &workers = owners_[actor->GetNodeID()];
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:740]: when will `owners_` be set?

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 4]
```diff
@@ -721,5 +721,36 @@ void GcsActorManager::SchedulePendingActors() {
...
+        auto it = workers.find(actor->GetWorkerID());
+        if (it == workers.end()) {
+          std::shared_ptr<rpc::CoreWorkerClientInterface> client =
+              worker_client_factory_(actor->GetOwnerAddress());
+          workers.emplace(actor->GetOwnerID(), Owner(std::move(client)));
+        }
+      }
+    }
+    RAY_LOG(INFO) << "Finished loading initial data.";
+    done(Status::OK());
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:750]: use an empty callback if the status is always ok.

File: src/ray/gcs/gcs_server/gcs_server.cc (hunk 1 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_server.cc]: fixed

File: src/ray/gcs/gcs_server/gcs_object_manager.h (hunk 2 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_object_manager.h]: fixed

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc (hunk 4 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_actor_manager.cc]: fixed

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc (hunk 3 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_actor_manager.cc:740]: owners_ will be set just here.

**Review by raulchen — APPROVED** [review]

**Pull request merged**